apps/src/applab/designMode.js

```diff
@@ -288,7 +288,9 @@ designMode.updateProperty = function (element, name, value) {
       var height = parseInt(element.style.height, 10);
       element.style.backgroundSize = width + 'px ' + height + 'px';
 
-      var url = ICON_PREFIX_REGEX.test(value) ? assetPrefix.renderIconToString(value, element) : assetPrefix.fixPath(value);
+      var url = ICON_PREFIX_REGEX.test(value) ?
+        assetPrefix.renderIconToString(value, element) :
+        assetPrefix.fixPath(encodeURIComponent(value));
```

Comment:
This is indeed trickier than I thought. One catch is that html attributes can have spaces in them while css urls cannot, but there's a little more to it than that. There are design elements for which we set image urls:

1. **image element** In this case, we use the `src` attribute on the `img` tag, so spaces are allowed.

2. **screen background** this is the broken case. we have to do something new here because spaces in css urls are not allowed. I initially proposed `encodeURIComponent`

3. **button element background** although this uses a css url, we first assign the url to an `Image` object and then fetch the `src` from it before assigning that to the CSS URL: https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/applab/designMode.js#L274-L276 so that's why we don't have to manually URI-encode.

Finally, encodeURIComponent won't work here because it will encode `/`, which will break any URLs with slashes in them, which was caught by the ec_screens -> "add a background" test. So, I've updated the tests to use the same trick shown for button images, and updated tests to confirm that it's working. I could move this trick inside `fixPath`, but I haven't done that because it's not necessary for Image elements.

In response to other things you commented on:

As for the 1x1.gif, that url appears because the css fails to update when you try to write invalid values to it, so you're left with whatever was there before.

`fixPath` is calling `encodeURIComponent` because that's necessary to send the image to the media proxy, which is a separate step from this.

Pleas take another look 👓